src/controller/java/templates/ClusterInfo-java.zapt

```diff
@@ -83,32 +195,78 @@ public class ClusterInfoMapping {
       }
 
   {{/chip_cluster_responses}}
+      {{#chip_server_cluster_attributes}}
+      {{#if isList}}
+      {{#if isStruct}}
+
+      {{/if}}
+
+      public class Delegated{{asUpperCamelCase name}}AttributeCallback implements ChipClusters.{{asUpperCamelCase ../name}}Cluster.{{asUpperCamelCase name}}AttributeCallback, DelegatedClusterCallback {
+        private ClusterCommandCallback callback;
+        @Override
+        public void setCallbackDelegate(ClusterCommandCallback callback) {
+          this.callback = callback;
+        }
+        @Override
+        public void onSuccess(List<
+        {{#if isStruct}}
+        ChipClusters.{{asUpperCamelCase ../name}}Cluster.{{asUpperCamelCase name}}Attribute
+        {{else}}
+          {{#if (isOctetString type)}}
+          byte[]
+          {{else if (isCharString type)}}
+          // Add String field here after ByteSpan is properly emitted in C++ layer
+          {{else}}
+          {{asJavaBasicTypeForZclType type true}}
+          {{/if}}
+        {{/if}}
+        > valueList) {
+          Map<CommandResponseInfo, Object> responseValues = new LinkedHashMap<>();
+          {{#if isStruct}}
+            CommandResponseInfo commandResponseInfo = new CommandResponseInfo("valueList", "List<ChipClusters.{{asUpperCamelCase ../name}}Cluster.{{asUpperCamelCase name}}Attribute>");
+          {{else}}
+            {{#if (isOctetString type)}}
+            CommandResponseInfo commandResponseInfo = new CommandResponseInfo("valueList", "List<byte[]>");
+            {{else if (isCharString type)}}
+            // Add String field here after ByteSpan is properly emitted in C++ layer
+            {{else}}
+            CommandResponseInfo commandResponseInfo = new CommandResponseInfo("valueList", "List<{{asJavaBasicTypeForZclType type true}}>");
+            {{/if}}
+          {{/if}}
+
+          responseValues.put(commandResponseInfo, valueList);
+          callback.onSuccess(responseValues);
+        }
+        @Override
+        public void onError(Exception ex) {
+          callback.onFailure(ex);
+        }
+      }
+      {{/if}}
+      {{/chip_server_cluster_attributes}}
+
   {{/chip_client_clusters}}
 
  public Map<String, ClusterInfo> getClusterMap() {
-     Map<String, ClusterInfo> clusterMap = new HashMap<>();
+    Map<String, ClusterInfo> clusterMap = new HashMap<>();
+    getCommandMap(clusterMap);
+    getReadAttributeMap(clusterMap);
+    return clusterMap;
```

Comment:
I think this design is cleaner:
```
Map<String, ClusterInfo> clusterMap = new HashMap<>();
Map<> commandMap = getCommandMap();
Map<> readMap = getReadAttributeMap();
combineCommands(clusterMap, commandMap);
combineCommands(clusterMap, readMap);

void combineCommands(Map<String, ClusterInfo> dest, Map<String, ClusterInfo> source) {
  // For every key-value pair, take the ClusterInfo commands from source and add them to the ClusterInfo 
  // commands in dest.
}
```
Since it moves the `combine` logic out of the individual functions.